src/OpenLoco/src/Map/TileManager.cpp

```diff
@@ -464,21 +464,26 @@ namespace OpenLoco::World::TileManager
         return height;
     }
 
-    SmallZ getSurfaceCornerHeight(const SurfaceElement& surface)
+    SmallZ getSurfaceCornerHeight(const SurfaceElement& surface, const uint8_t cornerMask)
     {
         auto baseZ = surface.baseZ();
-        if (surface.slope())
-        {
-            baseZ += kSmallZStep;
-        }
-        if (surface.isSlopeDoubleHeight())
+        if (surface.slope() & cornerMask)
         {
             baseZ += kSmallZStep;
+            if (surface.isSlopeDoubleHeight())
```

Comment:
i don't think this is the same. I would create a new function just for your code as it does something a bit more complex.
```cpp
if (surface.slope() & cornerMask)
{
    baseZ += kSmallZStep;
    if (surface.isSlopeDoubleHeight() && (surface.slope == oppositeCornerDown))
    {
        baseZ += kSmallZStep;
    }
}
```
where `oppositeCornerDown` is `SurfaceSlope::CornerDown::east` if `cornerMask` is `SurfaceSlope::CornerUp::west`